content/posts/messaging-infrastructure.md

```diff
@@ -0,0 +1,80 @@
+# Messaging infrastructure
+
+So far we've seen message de-duplication is not a simple topic and that *idempotency* is not a useful term then talking about distributed systems in which messages can get duplicated and reordered. Since we are sending our readers down the path of complexity we should at least discuss briefly what are the alternatives.
+
+## Building blocks
+
+In the model of a distributed system we use, all business processes are implemented by combining one simple building block. That building block consists of five operations
+- receive message
```

Comment:
Should we say a bit more about the order of these operations? It might be none obvious that this is pretty much the only reasonable one?